.github/workflows/mac.yml

```diff
@@ -0,0 +1,58 @@
+name: MacOS Build
+on:
+  pull_request:
+  push:
+    branches:
+      - master
```

Comment:
Should we run this on other branches as well, to get possible errors on PRs?

Reply:
The `pull_request` option right before `push` will allow this action to run on all PRs as well. But, the only branch where direct pushes without an associated PR will trigger this action is master.

I *think* the reason we used this configuration for the Linux action as well is because if we support other branches for `push` then a PR would trigger the action twice - once for the push and once for the PR. And the reason we wanted to specifically support trigger the action with PRs is so parts of the CI pipeline (i.e. the tests) could run if someone opens a PR from a fork of the repo.

Reply:
Didn't note the previous `pull_request` directive. This setup makes a lot of sense! Will try to do sth like that on the livepeer-data repo as well.